enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,504 @@
+the operator (same method and resultant actions as: https://github.com/openshift/hypershift/pull/1351) the README here 
+gives an AWS specific example: https://github.com/openshift/aws-pod-identity-webhook)
+
+Possibly some changes to logic for Pod Identity Webhooks as needed for fine-grained credential
+management to allow for multi-tenancy (see definition earlier). Also, need to add pod identity webhooks targeting other
+cloud providers and change the annotations to more generic naming.
+
+**HyperShift changes**: Include Cloud Credential Operator with "Token" mode (see above). Allows for processing of 
+CredentialsRequest objects added by operators.
+
+**OperatorHub and Console changes**: Allow for import of additional, well-known, bundle annotations which will result 
+in ENV variables on the Subscription object. Show operator is enabled for Token-based use based on Console determining 
+cluster is running in token-based authentication enabled cluster by reading the `.spec.serviceAccountIssuer` from the Authentication CR, `.spec.platformSpec.type` from the Infrastructure CR,`.spec.credentialsMode` from the CloudCredentials CR  AND
+bundle has token-based auth enabled annotation. This info should be written to a new field on Subscription objects.
+The Subscription fields will allow UX for the information needed by CCO or webhook, for input of the cloud credentials. 
+Generate a manual ack for operator upgrade when cloud resources are involved. This can be determined by parsing the 
+Subscription objects. Ensures admin signs off that cloud resources are provisioned and working as intended before an 
+operator upgrade.
+
+**Operator team/ Operator SDK changes**: Follow new guidelines for allowing for the operator to work on token auth 
+enabled cluster. New guidelines would include the following to use CCO Token mode:
+
+- changing CRD to add CredentialRequest references and putting those referenced
+- CredentialRequests into a defined directory in the bundle
+- Add a bundle annotation claiming token-based authentication support
+- add the projected ServiceAccount volume to deployment;
+- handle the Secret themselves (like today’s CCO Mint mode, read and use the credentials in the Secret whose name they 
+  know from the CredentialsRequest and is in the same Namespace)
+- Guidance to operator authors to structure controller code to alert on missing cloud credentials, not crash.
+
+SDK to support this new template. SDK to validate and in particular: alert on any permission changes between operator
+versions.
+
+### Workflow Description
+
+Making a layered operator ready to work on an STS enabled cluster will involve the following steps:
+
+For the operator author team:
+- Add CredentialRequests to the bundle, known location;
+- Add "role-arn"-type fields to the CredentialRequests
+- Add references to the CredentialRequests in the CRD spec;
```

Comment:
Will this be the trigger for OpenShift/OLM to distinguish if an operator needs credentials?

Reply:
I think that's the only reason we need this reference, so yes.

Reply:
isn't this only relevant if the operand needs creds?  At which point i feel like we're outside the scope of the problem.  (We got STS creds to the operator, from there it's up to the operator to get creds to the operand in whatever way it chooses)

Reply:
I'm taking this out because I agree it's beyond scope. But should we add a bundle annotation that is simply `tokenAwareEnabled`, or similar, for operator bundles? For the reason @jharrington22 gives, as a way to easily show in the UX that the operator needs setup for time-based token cluster (STS). @bparees ? @jharrington22 ?

Reply:
Removing "Add references to the CredentialRequests in the CRD spec;" in upcoming commit